frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -108,6 +112,18 @@ const RestorePVCModal = withHandlePromise<RestorePVCModalProps>(
         },
       };
 
+      if (pvcResource) {
+        // should set block only for RBD + RWX
+        if (
+          _.endsWith(snapshotClassResource?.driver, cephRBDProvisionerSuffix) &&
+          accessModes[0] === 'ReadWriteMany'
```

Comment:
I'm not as familiar with the PVC API, but we need to check if any of the array items are `ReadWriteMany`? Or that there is only one item in the array?

Reply:
@spadgett yes a PVC can have only 1 access mode at a time https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

Reply:
Hi, maybe I missed it, but I don't see where the link says there can only be one. (I'm also unsure why it would be an array in the API in that case.)

Reply:
As per my understanding: There can be multiple-access modes for a PV but when the PV is mounted using PVC there will be only one access mode that will be used. So we can assume that only one value will exist at a time. (I.e only fits for Static PV creation)

It's the same reason we have radio buttons for access mode while PVC creation.

Reply:
Hi @spadgett , the link says the `Important! A volume can only be mounted using one access mode at a time, even if it supports many.` But, PVC CR lets you create PVC with multiple access modes, so +1, we should handle it. Updated the PR to check in the array. Sorry for the confusion.